utils/sites.js

```diff
@@ -240,6 +240,32 @@ const updateParticipantData = async (req, res, authObj) => {
             continue;
         }
 
+        // Refect if the participant has withdrawn consent unless the update is
+        // for data distruction or hippa withdrawal 
+        const allowedWithdrawnUpdates = Object.keys(fieldMapping.dataDestruction);
+        const withdrawConsent = refusalWithdrawalConcepts.withdrewConsent.toString();
+        const revokeHIPAA = refusalWithdrawalConcepts.revokeHIPAA.toString();
+        if (docData[withdrawConsent] === fieldMapping.yes && docData[revokeHIPAA] === fieldMapping.yes) {
+            //Check the keys being sent in for the presence of allowed update
+            //For now if one acceptable key is found then the check will pass
+            //This could be made better and more secure with a comprehensive list of 
+            //all the keys allowed and make sure only those keys are allowed to be updated
+            let foundAcceptableKey = false;
+            let allowedUpdatesKeyIndex = 0;
+            while (!foundAcceptableKey || allowedUpdatesKeyIndex < allowedWithdrawnUpdates.length) {
+                if (Object.prototype.hasOwnProperty.call(dataObj, allowedWithdrawnUpdates[allowedUpdatesKeyIndex])) {
+                    foundAcceptableKey = true;
+                }
+                allowedUpdatesKeyIndex++;
```

Comment:
A contains check is slower than a hasOwnPropery check so it made more sense to me to check the dataObj for the existence of one of the expected keys than to check the dataObj keys against the list.  And I did a while loop to make it quicker to bail out of the loop instead of a forEach because I am only looking the presence of one of the keys